### PR TITLE
Store items display

### PR DESCRIPTION
**Pull request by malisara**

### CODE REVIEW

File: sir_cake/store/views.py [hunk 1]
```diff
@@ -48,9 +49,45 @@ def store(request):
         if redirect_ is not None:
             return redirect(redirect_)
 
+    context['items_in_basket'] = _get_preorder_or_none(request)
```
Comment (inejc) [inline sir_cake/store/views.py]: I suggest using a context processor for this: https://github.com/malisara/planty/search?q=context+processor. Also, I would rename the context key to something like `use_has_items_in_basket`. You can then pass in a boolean, e.g. `_get_preorder_or_none(...) is not None`

File: sir_cake/store/views.py [hunk 2]
```diff
@@ -108,7 +149,10 @@ def shopping_bag(request):
         BasketItem.objects.get(id=delete_pk).delete()
         messages.success(request, "Item deleted")
 
-    basket_items = BasketItem.objects.filter(order=preorder)
+        # Delete order if it there are no itemss
```
Comment (inejc) [inline sir_cake/store/views.py]: `Delete order if there are no items`

File: sir_cake/store/views.py [hunk 3]
```diff
@@ -108,7 +149,10 @@ def shopping_bag(request):
...
+        if basket_items.count() == 0:
```
Comment (inejc) [inline sir_cake/store/views.py]: Does this work? Shouldn't there still be a single item in `basket_items` since we deleted the last one after constructing `basket_items`? Also, I would set a `message` after deleting everything

File: sir_cake/store/views.py [hunk 4]
```diff
@@ -48,9 +49,45 @@ def store(request):
...
 
+    context['items_in_basket'] = _get_preorder_or_none(request)
     return render(request, 'store/store.html', context)
 
 
+def _available_items_and_category_context(request):
+    # Only items with inventory > 0 and inventory > all users' reservations
+    # are returned
+
+    total_quantity_bag_items = BasketItem.objects.filter(
```
Comment (inejc) [inline sir_cake/store/views.py]: Maybe `total_quantity_basket_items`?

File: sir_cake/store/views.py [hunk 5]
```diff
@@ -48,9 +49,45 @@ def store(request):
...
+        item_to_buy_id=OuterRef('pk')).values(
+            'item_to_buy_id').annotate(total=Sum('quantity')).values('total')
+
+    items_not_in_basket = Item.objects.filter(quantity__gte=0).exclude(
+        id__in=BasketItem.objects.values_list('item_to_buy', flat=True))
+
+    items_in_basket_enough_quantity = Item.objects.filter(
+        quantity__gt=Subquery(total_quantity_bag_items))
+
+    items = items_in_basket_enough_quantity.order_by(
```
Comment (inejc) [inline sir_cake/store/views.py]: Should we do order by after merging them with `|`?

File: sir_cake/store/views.py [hunk 6]
```diff
@@ -48,9 +49,45 @@ def store(request):
...
+        '-id') | items_not_in_basket.order_by('-id')
+
+    category = request.GET.get('category')
+    if category != "" and category is not None and category != 'ALL':
+        items = items.filter(category=category)
```
Comment (inejc) [inline sir_cake/store/views.py:75]: What if `category` is an invalid value like `1324afa`?

File: sir_cake/store/views.py (hunk 3 above)
Comment (malisara) [reply sir_cake/store/views.py]: It works :). According to Django documentation are Querysets lazy: 

> In general, the results of a [QuerySet] aren’t fetched from the database until you “ask” for them. When you do, the QuerySet is evaluated by accessing the database.

File: sir_cake/store/utils.py [hunk 7]
```diff
@@ -30,3 +32,21 @@ def get_items_and_prices_and_order_sum(shopping_bag):
         'items_and_prices': items_and_prices,
         'total_price_all_items': total_order_price(shopping_bag),
     }
+
+
+def get_preorder_or_none(request):
```
Comment (inejc) [inline sir_cake/store/utils.py:42]: I suggest putting `get_preorder_or_none` after the context processor func

File: sir_cake/store/views.py [hunk 8]
```diff
@@ -68,7 +107,7 @@ def store_item_detail(request, pk):
 
     form = BasketItemForm(_get_max_quantity_to_buy(item))
     return render(request, 'store/store-item-detail.html',
-                  {'item': item, 'form': form})
+                  {'item': item, 'form': form, })
```
Comment (inejc) [inline sir_cake/store/views.py]: Why this change?

File: sir_cake/store/views.py [hunk 9]
```diff
@@ -51,6 +53,43 @@ def store(request):
...
+        id__in=BasketItem.objects.values_list('item_to_buy', flat=True))
+
+    items_in_basket_enough_quantity = Item.objects.filter(
+        quantity__gt=Subquery(total_quantity_basket_items))
+
+    items = items_in_basket_enough_quantity | items_not_in_basket
+
+    category = request.GET.get('category')
+
+    if category in dict(Item.CATEGORIES_CHOICES) \
```
Comment (inejc) [inline sir_cake/store/views.py]: `SHORT_CATEGORY_TO_NAME` is already a dict you need :)
